lib/models/users/matrix.js

```diff
@@ -69,6 +75,23 @@ MatrixUser.prototype.serialize = function() {
     this._data.localpart = this.localpart;
     return this._data;
 };
-
+/**
+ * Make a userId conform to the matrix spec using QP escaping.
+ * Grammar taken from: https://matrix.org/docs/spec/appendices.html#identifier-grammar
+ */
+MatrixUser.prototype.escapeUserId = function() {
+    // Currently Matrix accepts / in the userId, although going forward it will be removed.
+    const badChars = new Set(this.localpart.replace(/([A-z0-9]|-|\.|=|_)+/g, ""));
+    let res = this.localpart;
+    badChars.forEach((c) => {
+        const hex = c.charCodeAt(0).toString(16).toUpperCase();
```

Comment:
wouldn't we want to lowercase the hex?

Reply:
I prefer uppercase hex, but I don't know if we have a standard for it.